### ISSUE TITLE
Linux permissions don't appear to be set correctly on update

### ISSUE THREAD

**Issue by unpacklo** [itchio/itch#655]
From twitter: https://twitter.com/Roflraging/status/731270224692031489

> Don't know if it's version specific, but I updated Brigador and grabbed update through itch app Linux and couldn't launch.
> 
> after some sleuthing, discovered that the binary didn't have the executable permission set.  Once I set it, it could launch.
> 
> Pushed directly through butler and original files on my disk had permissions set correctly.
> 
> Did I do something wrong or did I hit a bug somewhere?
> 
> Completely uninstalling and reinstalling through itch app works fine though.

I pushed from my local machine:

```
dalekim1@lispy:/home/dalekim1/itchio> ./butler --version
head, built on May  8 2016 @ 08:39:40
```

And my itch app was updated yesterday using itch_17.6.0_amd64.deb.

I will try updating butler and see what happens!

**Comment by unpacklo** [comment]
I just tried updating and either I'm already on the most up to date version or it didn't work?

```
dalekim1@lispy:/home/dalekim1/itchio> ./butler --version
head, built on May  8 2016 @ 08:39:40
dalekim1@lispy:/home/dalekim1/itchio> ./butler update
✓ Bleeding-edge, not upgrading to stable unless told to.
(Use `--head` if you want to upgrade to the latest bleeding-edge version)
dalekim1@lispy:/home/dalekim1/itchio> ./butler update --head
:: Do you want to upgrade to the bleeding-edge version? Things may break! [y/N] y
• https://dl.itch.ovh/butler/linux-amd64/head/butler.gz
Downloading 5.1 MB from cloudflare-nginx at dl.itch.ovh
• Backing up current version to /home/dalekim1/itchio/butler.old just in case...
✓ Upgraded butler from head to head. Have a nice day!
dalekim1@lispy:/home/dalekim1/itchio> ./butler --version
head, built on May  8 2016 @ 08:39:40
dalekim1@lispy:/home/dalekim1/itchio> ls
brigadorlinux  butler  butler.new.gz  public_builds
dalekim1@lispy:/home/dalekim1/itchio> ls butler.new.gz 
brigadorlinux/ butler         butler.new.gz  public_builds/ 
dalekim1@lispy:/home/dalekim1/itchio> ll .
total 22920
drwxr-xr-x  3 dalekim1 dalekim1     4096 May 14 17:36 .
drwxr-xr-x 80 dalekim1 dalekim1     4096 May 14 17:25 ..
drwxr-xr-x  8 dalekim1 dalekim1     4096 May 14 17:31 brigadorlinux
-rwxr-xr-x  1 dalekim1 dalekim1 18385650 May 14 17:36 butler
-rw-r--r--  1 dalekim1 dalekim1  5069625 May 14 17:36 butler.new.gz
lrwxrwxrwx  1 dalekim1 dalekim1       17 May 13 18:35 public_builds -> ../public_builds/
dalekim1@lispy:/home/dalekim1/itchio> md5sum butler
1a3d7b9d08edf849992d7bed4de8daa4  butler
```

Is this expected?  As you can see there is no 'butler.old' anywhere, but I unpacked butler.new and it seems like it's the same as the one I have in the directory now, so I guess it did still update (or I already had that version)?

**Comment by quyse** [comment]
I see that too, I created a test game project on itch and pushed first version with Linux executable via butler, downloaded it via itch app, then pushed the second version with slightly changed Linux executable, then updated it via itch app. After initial installation the game runs fine, but after upgrade it fails. Indeed the executable's permissions after update is wrong: `-rw-r--r--`. Removing and re-downloading game in itch app fixes that, but another upgrade breaks it again. Checked with latest itch from AUR, and itch compiled manually from `master` branch.
BTW, initial permissions for executable (after first installation) are not great too: `-rwxrwxrwx`, so it's writable by any user in the system. Probably it should be `-rwxr-xr-x`.
